logstash-core/src/main/java/org/logstash/config/ir/compiler/DatasetCompiler.java

```diff
@@ -490,9 +565,13 @@ public Collection<JrubyEventExtLibrary.RubyEvent> compute(@SuppressWarnings("raw
             if (done) {
                 return data;
             }
-            parent.compute(batch, flush, shutdown);
-            done = true;
-            return data;
+            try {
+                parent.compute(batch, flush, shutdown);
+                done = true;
+                return data;
+            } catch (ConditionalEvaluationError ex) {
+                return data;
+            }
```

Comment:
This is the complement branch (the right one, the `else` of the empty branch of an `if`). The dataset execute positive path on if and the complement. If an evaluation error occur in EventCondition it will happen also in the complementary evaluation of the same event condition. So silence the error, the data remains empty so the execution flow interrupts here for the batch.